src/Sylius/Bundle/ShopBundle/Resources/config/app/twig_hooks/base.yaml

```
@@ -29,12 +29,58 @@ twig_hooks:
+                
+        'sylius_shop.base.header.content.security':
+            logged_in_user:
+                template: '@SyliusShop/shared/layout/base/header/content/security/logged_in_user.html.twig'
+            visitor:
+                template: '@SyliusShop/shared/layout/base/header/content/security/visitor.html.twig'
+                
+        'sylius_shop.base.header.content.security.logged_in_user':
+            mobile:
+                template: '@SyliusShop/shared/layout/base/header/content/security/logged_in_user/mobile.html.twig'
+            desktop:
+                template: '@SyliusShop/shared/layout/base/header/content/security/logged_in_user/desktop.html.twig'
+                
+        'sylius_shop.base.header.content.security.logged_in_user.mobile':
+            dropdown_button:
+                template: '@SyliusShop/shared/layout/base/header/content/security/logged_in_user/mobile/dropdown_button.html.twig'
+            dropdown_menu:
+                template: '@SyliusShop/shared/layout/base/header/content/security/logged_in_user/mobile/dropdown_menu.html.twig'
+
+        'sylius_shop.base.header.content.security.logged_in_user.mobile.dropdown_menu':
+            dashboard_button:
+                template: '@SyliusShop/shared/layout/base/header/content/security/logged_in_user/mobile/dropdown_menu/dashboard_button.html.twig'
+            logout_button:
+                template: '@SyliusShop/shared/layout/base/header/content/security/logged_in_user/mobile/dropdown_menu/logout_button.html.twig'
+
+        'sylius_shop.base.header.content.security.logged_in_user.desktop':
+            welcome_message:
+                template: '@SyliusShop/shared/layout/base/header/content/security/logged_in_user/desktop/welcome_message.html.twig'
+            dashboard_button:
+                template: '@SyliusShop/shared/layout/base/header/content/security/logged_in_user/desktop/dashboard_button.html.twig'
+            logout_button:
+                template: '@SyliusShop/shared/layout/base/header/content/security/logged_in_user/desktop/logout_button.html.twig'
+        
+        'sylius_shop.base.header.content.security.visitor':
+            mobile:
+                template: '@SyliusShop/shared/layout/base/header/content/security/visitor/mobile.html.twig'
+            desktop:
+                template: '@SyliusShop/shared/layout/base/header/content/security/visitor/desktop.html.twig'
+        
+        'sylius_shop.base.header.content.security.visitor.desktop':
+            user_icon:
+                template: '@SyliusShop/shared/layout/base/header/content/security/visitor/desktop/user_icon.html.twig'
+            login_button:
+                template: '@SyliusShop/shared/layout/base/header/content/security/visitor/desktop/login_button.html.twig'
+            register_button:
+                template: '@SyliusShop/shared/layout/base/header/content/security/visitor/desktop/register_button.html.twig'
```

```suggestion
            login:
                template: '@SyliusShop/shared/layout/base/header/content/security/visitor/desktop/login_button.html.twig'
            register:
                template: '@SyliusShop/shared/layout/base/header/content/security/visitor/desktop/register_button.html.twig'
```
What's with all the suffixes?